### DISCUSSION TITLE
onPreInit not showing all types

### DISCUSSION THREAD

**Discussion by wahidshafique** [gatsbyjs/gatsby#35689] [Help] [answered]
I get ts red underlines on this bit of code
```
export const onPreInit = async ({
  actions,
  store,
}: GatsbyNode['onPreInit']) => {
```
`Property 'actions' does not exist on type '((args: PreInitArgs, options: PluginOptions, callback: PluginCallback<void>) => void | Promise<void>) | undefined'.ts(2339)`

But according to code I've worked on and that works for other [people](https://stackoverflow.com/a/65957062/7667248) these objects should exist in the signature. Anyone know what I'm doing wrong?

**Comment by krismorf** [comment, marked as answer]
@wahidshafique from the [docs](https://www.gatsbyjs.com/docs/how-to/custom-configuration/typescript/#gatsby-nodets) it should be like below:
```tsx
export const onPreInit: GatsbyNode["onPreInit"] = async ({
  actions,
  store,
}) => {
  //  your code
}
```
